File: app/src/main/java/com/asksira/loopingviewpagerdemo/MainActivity.java
package com.asksira.loopingviewpagerdemo;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.asksira.loopingviewpager.LoopingViewPager;
import com.rd.PageIndicatorView;

import java.util.ArrayList;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    LoopingViewPager viewPager;
    DemoInfiniteAdapter adapter;
    PageIndicatorView indicatorView;
    Button changeDataSetButton;

    TextView changePageLabel;
    Button page1;
    Button page2;
    Button page3;
    Button page4;
    Button page5;
    Button page6;

    private int currentDataSet = 1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        viewPager = findViewById(R.id.viewpager);
        indicatorView = findViewById(R.id.indicator);
        changeDataSetButton = findViewById(R.id.change_dataset);
        changeDataSetButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                changeDataset();
            }
        });

        changePageLabel = findViewById(R.id.change_page_label);
        page1 = findViewById(R.id.page1);
        page2 = findViewById(R.id.page2);
        page3 = findViewById(R.id.page3);
        page4 = findViewById(R.id.page4);
        page5 = findViewById(R.id.page5);
        page6 = findViewById(R.id.page6);

        adapter = new DemoInfiniteAdapter(this, createDummyItems(), true);
        viewPager.setAdapter(adapter);

        //Custom bind indicator
        indicatorView.setCount(viewPager.getIndicatorCount());
        viewPager.setIndicatorPageChangeListener(new LoopingViewPager.IndicatorPageChangeListener() {
            @Override
            public void onIndicatorProgress(int selectingPosition, float progress) {
                indicatorView.setProgress(selectingPosition, progress);
            }

            @Override
            public void onIndicatorPageChange(int newIndicatorPosition) {
//                indicatorView.setSelection(newIndicatorPosition);
            }
        });

        View.OnClickListener pageSelector = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Integer number = Integer.valueOf(((Button)v).getText().toString());
                viewPager.setCurrentItem(adapter.isInfinite() ? number : number - 1);
            }
        };

        page1.setOnClickListener(pageSelector);
        page2.setOnClickListener(pageSelector);
        page3.setOnClickListener(pageSelector);
        page4.setOnClickListener(pageSelector);
        page5.setOnClickListener(pageSelector);
        page6.setOnClickListener(pageSelector);
    }

    @Override
    protected void onResume() {
        super.onResume();
        viewPager.resumeAutoScroll();
    }

    @Override
    protected void onPause() {
        viewPager.pauseAutoScroll();
        super.onPause();
    }

    private ArrayList<Integer> createDummyItems () {
        ArrayList<Integer> items = new ArrayList<>();
        items.add(0, 1);
        items.add(1, 2);
        items.add(2, 3);
        items.add(3, 4);
        items.add(4, 5);
        items.add(5, 6);
        items.add(6, 0);
        return items;
    }

    private ArrayList<Integer> createSecondDummyItems() {
        ArrayList<Integer> items = new ArrayList<>();
        items.add(0, 1);
        items.add(1, 2);
        return items;
    }

    private ArrayList<Integer> createThirdDummyItems() {
        ArrayList<Integer> items = new ArrayList<>();
        items.add(0, 1);
        return items;
    }

    private void changeDataset () {
        if (currentDataSet == 1) {
            adapter.setItemList(createSecondDummyItems());
            currentDataSet++;
            toggleSixButtons(false);
        } else if (currentDataSet == 2) {
            adapter.setItemList(createThirdDummyItems());
            currentDataSet++;
            toggleSixButtons(false);
        } else {
            adapter.setItemList(createDummyItems());
            currentDataSet = 1;
            toggleSixButtons(true);
        }
        indicatorView.setCount(viewPager.getIndicatorCount());
        viewPager.reset();
    }

    private void toggleSixButtons (boolean isVisible) {
        int status = isVisible ? View.VISIBLE : View.GONE;
        changePageLabel.setVisibility(status);
        page1.setVisibility(status);
        page2.setVisibility(status);
        page3.setVisibility(status);
        page4.setVisibility(status);
        page5.setVisibility(status);
        page6.setVisibility(status);
    }
}

File: app/src/main/java/com/asksira/loopingviewpagerdemo/MainActivity.kt
package com.asksira.loopingviewpagerdemo

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.asksira.loopingviewpagerdemo.databinding.ActivityMainBinding
import com.asksira.loopingviewpagerdemo.databinding.ItemSelectedBinding
import com.asksira.loopingviewpagerdemo.databinding.ItemUnselectedBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private var adapter: DemoInfiniteAdapter? = null
    private var currentDataSet = 1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        binding.changeDataset.setOnClickListener { changeDataset() }
        binding.btnListActivity.setOnClickListener {
            startActivity(Intent(this, ListActivity::class.java))
        }


        adapter = DemoInfiniteAdapter(createDummyItems(), true)
        binding.viewpager.adapter = adapter

        binding.viewpager.onIndicatorProgress = { selectingPosition, progress ->
            binding.indicator.onPageScrolled(
                selectingPosition,
                progress
            )
        }
        setupClicks()
        binding.indicator.updateIndicatorCounts(binding.viewpager.indicatorCount)
    }

    private fun setupClicks() {
        val pageSelectorListener =
            View.OnClickListener { v ->
                val number =
                    Integer.valueOf((v as Button).text.toString())
                binding.viewpager.currentItem =
                    if (adapter?.isInfinite == true) number else number - 1
            }
        binding.page1.setOnClickListener(pageSelectorListener)
        binding.page2.setOnClickListener(pageSelectorListener)
        binding.page3.setOnClickListener(pageSelectorListener)
        binding.page4.setOnClickListener(pageSelectorListener)
        binding.page5.setOnClickListener(pageSelectorListener)
        binding.page6.setOnClickListener(pageSelectorListener)
    }

    override fun onResume() {
        super.onResume()
        binding.viewpager.resumeAutoScroll()
    }

    override fun onPause() {
        binding.viewpager.pauseAutoScroll()
        super.onPause()
    }

    private fun changeDataset() {
        if (currentDataSet == 1) {
            adapter?.itemList = createSecondDummyItems().toList()
            currentDataSet++
            toggleSixButtons(false)
        } else if (currentDataSet == 2) {
            adapter?.itemList = createThirdDummyItems().toList()
            currentDataSet++
            toggleSixButtons(false)
        } else {
            adapter?.itemList = createDummyItems().toList()
            currentDataSet = 1
            toggleSixButtons(true)
        }
        binding.indicator.updateIndicatorCounts(binding.viewpager.indicatorCount)
        binding.viewpager.reset()
    }

    private fun toggleSixButtons(isVisible: Boolean) {
        val status = if (isVisible) View.VISIBLE else View.GONE
        binding.changePageLabel.visibility = status
        binding.page1.visibility = status
        binding.page2.visibility = status
        binding.page3.visibility = status
        binding.page4.visibility = status
        binding.page5.visibility = status
        binding.page6.visibility = status
    }
}